### PR TITLE
Chat with GitHub Frontend UI

### LINKED ISSUE #148: chat with github repository
we want to build an application similar to [this one](https://github.com/ploomber/doc/tree/main/examples/panel/chat-with-github) but using FastAPI (we'll only build the API for now, we'll tackle the frontend later)

there's a few endpoints that this app needs:

and endpoint to parse the contents of a repo `POST /scrape/` should scrape the data from a github repo (should take `repo` in the body request. eg `ploomber/jupysql`) and return a repo_id

`GET /status/{repo_id}` should return the status of a repo_id: `pending`, `finished` to check whether scraping has finished

`POST /ask/` should contain a `question` and `repo_id` in the body, should return the answer to the question using that repository

since scraping will take a minute or so, we need to implement a task queue to run jobs in the background, I think celery is the simplest option

important: everything has to be prepared in a single dockerfile

### PR DESCRIPTION
**Pull request by bryannho**
### Changes

- Updates `chat-with-github-api` example to include Solara frontend
- Reorganizes app into `api/` and `client/` folders
- Updates `Dockerfile` to deploy all processes in one container

Closes #148 

----
📚 Documentation preview 📚: https://ploomber-doc--170.org.readthedocs.build/en/170/

### CODE REVIEW

File: examples/fastapi/chat-with-github-api/client/main.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+def Page():
+    sl.Style(css)
+    with sl.VBox(classes=["main"]) as ch:
+        Chat()
+    
+    return ch
+    # with sl.Column(align="start", classes=["plug"]):
+    #     with sl.HBox(align_items="center"):
+    #         sl.Image(f"static/system-logo.png", classes=["logo"])
+    #         sl.HTML(tag="a", unsafe_innerHTML="Built by Ploomber", attributes={"href": "https://ploomber.io", "target": "_blank"}, classes=["link"])
```
Comment (edublancas) [inline examples/fastapi/chat-with-github-api/client/main.py:45]: ensure you remove commented code

File: examples/fastapi/chat-with-github-api/requirements.txt [hunk 2]
```diff
@@ -8,4 +8,6 @@ sqlalchemy
 llama-index-core
 llama-index-readers-github
 llama_index
-humanize
\ No newline at end of file
+humanize
+solara==1.29.1
+openai
```
Comment (edublancas) [inline examples/fastapi/chat-with-github-api/requirements.txt:13]: provide specific versions so the app doesn't break with upgrades

**Review by edublancas — CHANGES_REQUESTED** [review]

File: examples/fastapi/chat-with-github-api/requirements.txt (hunk 2 above)
Comment (bryannho) [reply examples/fastapi/chat-with-github-api/requirements.txt]: [added versions](https://github.com/ploomber/doc/pull/170/files#diff-21fe5dc251f728d09d24529cc8556e342bffd8e35528ebac4ee072c8a0334b3cR2-R13)

**Comment by edublancas** [comment]
@bryannho are we linking from the blog post to the example? if not, please add a link

**Comment by bryannho** [comment]
> @bryannho are we linking from the blog post to the example? if not, please add a link

@edublancas the blog post links to the example code but not the deployed app. do you want me to add a link to the deployed app?

**Comment by edublancas** [comment]
no, it's ok. i meant a link to the example code

**Pull request merged**